Docs: Genesis file header info (#1422)
ava-labs/avalanche-docs: docs/subnets/customize-a-subnet.md

#### Fee Config

> `gasLimit`: Sets the max amount of gas consumed per block. Keep in mind that this will limit how 
> much computation can be processed in a single block and therefore will set an upper limit for 
> the maximum amount of gas that a single transaction can use.

Reviewer: ```suggestion
`gasLimit`: Sets the max amount of gas consumed per block. This restriction puts a cap on the
amount of computation that can be done in a single block, which in turn sets a limit on the 
maximum gas usage allowed for a single transaction.
```

For reference, C-Chain value is set to `15,000,000`.

`targetBlockRate`: Sets the target rate of block production in seconds. A target of 2 will target

> producing a block every 2 seconds. If the network starts producing faster than this, means that 
> more blocks than expected are being issued to the network, so the base fees will
> increased accordingly.

Reviewer: ```suggestion
producing a block every 2 seconds. If the network starts producing blocks at a faster rate, it 
indicates that more blocks than anticipated are being issued to the network, resulting in an 
increase in base fees.
```

set to match the `gasLimit` set in the `feeConfig`. You do not need to change any of the other genesis
header fields.

`nonce`, `mixHash` and `difficulty`

These are remnant parameters from Proof of Work systems.
For Avalanche, these don't play any relevant role, so you should just leave them as: 

- `nonce` as `0x0`.
- `mixHash` as `0x0000000000000000000000000000000000000000000000000000000000000000`.

> - `diffuculty` as `0x0.

Reviewer 2: ```suggestion
- `difficulty` as `0x0`.
```

> - `diffuculty` as `0x0.

Reviewer 2: Why we skipped explaining these? I know they're not very significant but again none of these Header fields actually matter and they should be left as they're (in most cases). So I feel like if other header fields need explanation, these 3 fields also should be explained too.

`timestamp`

The timestamp of the creation of the genesis block. This is commonly set to `0x0`.

`extraData`

Optional extra data that can be included in the genesis block. This is commonly set to `0x`.

`gasLimit`

The total amount of gas that can be used in a single block. It should be set to
the same value as in the [fee config](#fee-config). The value `e4e1c0` is
hexadecimal and is equal to `15,000,000`.

`coinbase`

Refers to a special transaction included in each block of the blockchain. It is the first 
transaction within the block and serves as the mechanism through which the block 
reward is generated. It is usually set

> to `0x0000000000000000000000000000000000000000` for the genesis block.

Reviewer 2: This sounds like `coinbase` itself is the transaction. `coinbase` is an recipient address for block rewards (in our case only tx fees) that could be collected by block producers. In other words this is the address of whomever produced the block (i.e miner)

Author: You're right! it does sound like that. For Avalanche, I understand there is no a single "miner" for each block, so for every block other than genesis, how does this works? what address its registered there?

Reviewer 2: Blocks are still proposed/built by single entities (miner/producer/proposer etc). 

The block structure of Coreth and Subnet-EVM are very similar to Eth blocks. Coinbase is always a part of coreth and subnet-evm blocks and represent the rewards address of whomever created the block. In Coreth this address has to be a blackhole address meaning that we burn all fees. However in Subnet-EVM there is an optional config (`allowFeeRecipients`) that can re-enable distributing these fees to the block producers through the coinbase. So in Subnet-EVM the coinbase can still represent block producer addresses.

`parentHash`

This is the Keccak 256-bit hash of the entire parent block’s header. It is
usually set to
`0x0000000000000000000000000000000000000000000000000000000000000000` for the
genesis block.

`gasUsed`

This is the amount of gas used by the genesis block. It is usually set to `0x0`.

`number`

> This is the number of the genesis block. It is usually set to `0x0`.

Reviewer 2: This required to be `0x0` for the genesis. Otherwise it will error.

> 

Reviewer 2: extra whitespaces here

set to match the `gasLimit` set in the `feeConfig`. You do not need to change any of the other genesis
header fields.

`nonce`, `mixHash` and `difficulty`

These are remnant parameters from Proof of Work systems.
For Avalanche, these don't play any relevant role, so you should just leave them as their 
default values:

`nonce` 

The result of the mining process iteration is this value. It can be any value in 
the genesis block. Default value is `0x0`.

`mixHash` 
The combination of `nonce` and `mixHash` allows to verify that the Block has really been 
cryptographically mined, thus, from this aspect, is valid. Default value is `0x0000000000000000000000000000000000000000000000000000000000000000`.

`difficulty`

The difficulty level applied during the nonce discovering process of this block. Default value is `0x0`.

> `timestamp`
> 
> The timestamp of the creation of the genesis block. This is commonly set to `0x0`.

Reviewer: Please adjust all param definitions to match throughout the doc. For example:

```suggestion
`timestamp`: The timestamp of the creation of the genesis block. This is commonly set to `0x0`.
```

`coinbase`

> Refers to the address of the recipient of the block reward. This is included in the first 
> transaction of each block of the blockchain. It is usually set

Reviewer 2: ```suggestion
Refers to the address of the block producers. This also means it represents the recipient of the block reward. It is usually set
```
